### DISCUSSION TITLE
USART DMA Receive - cyclic check

### DISCUSSION THREAD

**Discussion by RaZVaNx17** [MaJerle/STM32_USART_DMA_RX#27] [General]
Hi,
First of all thank you for all your work. It's very useful.

I am using a Nucleo F401RE with HAL library.

I configured my DMA cu give me a interrupt for IDLE and FT.
I  want to be able to come every 5ms (cyclic) to check if there is any data received. If there is, I want to process it and restart the DMA.
But in case the buffer gets full in the 5ms cycle time that I have set for it, I would like the DMA to notify me by using the FT interrupt.

How can I check the state of the buffer periodically when configured to use DMA?

Thank you again!

**Comment by MaJerle** [comment]
The trick is that you shouod never exit from dma rx mode. Thats where corcular mode comes to play.

Open oje example and check function that reads data. Typical name is 'check_rx'

You also need HT interrupt.

**Reply by RaZVaNx17** [reply]
Thank you for the quick response. I will try it out!
